src/compiler.js

```diff
@@ -204,3 +211,73 @@ SelectFormat.prototype.getOption = function (value) {
     var options = this.options;
     return options[value] || options.other;
 };
+
+function ShortNumberFormat(locales, options) {
```

Comment:
Could be its own lib so changes to it are not necessarily coupled to the compiler.  Can be a future PR.

For now, I'd move into its own module though.

Reply:
No this seems fine for now